src/wampy.js

```diff
@@ -903,7 +903,35 @@ class Wampy {
             return this._hardClose(errorURI, `Received message "${messageType}" before session was established`);
         }
 
-        await handler();
+        if (this._isRequestIdValid(data)) {
+            await handler();
+        }
+    }
+
+    /**
+     * Validates the requestId for message types that need this kind of validation
+     * @param {Array} data - [messageType, requestId]
+     * @return {boolean} true if it's a valid request and false if it isn't
+     * @private
+     */
+    _isRequestIdValid ([messageType, requestId]) {
+        const isRequestIdValidationNeeded = [
+            WAMP_MSG_SPEC.SUBSCRIBED,
+            WAMP_MSG_SPEC.UNSUBSCRIBED,
+            WAMP_MSG_SPEC.PUBLISHED,
+            WAMP_MSG_SPEC.RESULT,
+            WAMP_MSG_SPEC.REGISTERED,
+            WAMP_MSG_SPEC.UNREGISTERED
+        ].includes(messageType);
+        const isResultMessage = messageType === WAMP_MSG_SPEC.RESULT;
+
+        if (isRequestIdValidationNeeded
+            && (isResultMessage && !this._calls[requestId]
```

Comment:
Really hard to read these checks. At least they should be grouped with `()`

Reply:
Decided to split them into three different IF blocks for readability